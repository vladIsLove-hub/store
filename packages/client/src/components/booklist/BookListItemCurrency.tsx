import React from 'react';

import { Discount } from './BookListTypes';

const mainCurrency = ' $';

const BookListItemCurrency = ({ price, discount }: Discount) => {
  return (
    <div data-testid="currency" className="book-item__prices">
      <span className="book-item__price">
        {getActualPrice(price, discount)}
        {mainCurrency}
      </span>
      {discount !== 0 && (
        <>
          <span className="book-item__overprice">
            {price}
            {mainCurrency}
          </span>
          <span className="book-item__percent">-{(discount * 100).toFixed(1)} %</span>
        </>
      )}
    </div>
  );
};

function getActualPrice(price: number, percent: number): number {
  const discount = Number((price * percent).toFixed(3));
  return Number((price - discount).toFixed(2));
}

export default BookListItemCurrency;
