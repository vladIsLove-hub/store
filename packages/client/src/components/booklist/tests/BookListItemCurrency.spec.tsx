import React from 'react';
import { render } from '@testing-library/react';

import BookListItemCurrency from '../BookListItemCurrency';

describe('BookListItemCurrency', () => {
  it('should render correctly', () => {
    const { getByText, getByTestId } = render(<BookListItemCurrency discount={0.11} price={11} />);
    const container = getByTestId('currency');

    expect(container).toContainElement(getByText('9.79 $'));
    expect(container).toContainElement(getByText('11 $'));
    expect(container).toContainElement(getByText('-11.0 %'));
  });

  it('should render correctly without discount', () => {
    const { getByText, getByTestId } = render(<BookListItemCurrency discount={0} price={11} />);
    const container = getByTestId('currency');

    expect(container).toContainElement(getByText('11 $'));
  });

  it('BookListItemCurrency', () => {
    const component = render(<BookListItemCurrency discount={0.11} price={11} />);
    expect(component).toMatchSnapshot();
  });
});
