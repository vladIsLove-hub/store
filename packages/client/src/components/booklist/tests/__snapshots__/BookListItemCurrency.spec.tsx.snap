// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookListItemCurrency BookListItemCurrency 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="book-item__prices"
        data-testid="currency"
      >
        <span
          class="book-item__price"
        >
          9.79
           $
        </span>
        <span
          class="book-item__overprice"
        >
          11
           $
        </span>
        <span
          class="book-item__percent"
        >
          -
          11.0
           %
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="book-item__prices"
      data-testid="currency"
    >
      <span
        class="book-item__price"
      >
        9.79
         $
      </span>
      <span
        class="book-item__overprice"
      >
        11
         $
      </span>
      <span
        class="book-item__percent"
      >
        -
        11.0
         %
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
